lib/elixir/test/elixir/kernel_test.exs

```diff
@@ -502,6 +502,26 @@ defmodule KernelTest do
 
       assert r == 0
     end
+
+    test "calling if with invalid keys" do
+      assert_raise ArgumentError,
+        "invalid key `foo` in if, valid keys are `do` and `else`",
+        fn ->
+          Code.eval_string("if true, foo: 7")
+        end
+
+      assert_raise ArgumentError,
+        "invalid key `doo` in if, valid keys are `do` and `else`",
+        fn ->
+          Code.eval_string("if true, doo: 7, else: 6")
+        end
+
+      assert_raise ArgumentError,
+        "duplicate `do` key in if",
+      fn ->
```

Comment:
We usually bind the message to a variable and call `assert_raise` like this `assert_raise ArgumentError, message, fn -> ...` if it doesn't on one line.